### PR TITLE
Reduce to epochs

### PR DESCRIPTION
**Pull request by fraimondo**
Allows to reduce to epochs, creating a dictionary from marker -> list where each element of the list is the marker value for that specific epoch.

### CODE REVIEW

File: nice/collection.py [hunk 1]
```diff
@@ -72,6 +72,28 @@ def topo_names(self):
     def scalar_names(self):
         return list(self.keys())
 
+    def reduce_to_epochs(self, marker_params):
```
Comment (dengemann) [inline nice/collection.py:75]: Can we start adding doc strings? This way we have to do less work later.

File: nice/markers/base.py [hunk 2]
```diff
@@ -118,6 +118,10 @@ def _reduce_to(self, reduction_func, target, picks):
             out = func(out, axis=0)
         return out
 
+    def reduce_to_epochs(self, reduction_func, picks=None):
```
Comment (dengemann) [inline nice/markers/base.py:121]: Also here.

**Comment by dengemann** [comment]
Can we make an example? Or can it be that this is actually useful for any of the existing examples. I think the motor imagery one perhaps.

**Comment by dengemann** [comment]
@fraimondo cool for the docs. Btw. there is this vscode private file. I think it should not be committed.

**Comment by fraimondo** [comment]
What about now? Can we merge and add an issue for the example?

**Comment by dengemann** [comment]
@fraimondo there is still the vscode settings json. Can we remove  that?

**Comment by fraimondo** [comment]
I removed it.

It is not there anymore.

**Comment by dengemann** [comment]
Cool! Thanks.

**Pull request merged**